### PR TITLE
Armos filesystem rework

### PR DESCRIPTION
**Pull request by GermanHydrogen**
Closing #82 
Closing #83 
Closing #87 
Closing #74

### CODE REVIEW

File: addons/filesystem/functions/fnc_chdir.sqf [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/**
+ * Change directory.
+ *
+ * Arguments:
+ * 1: Pointer <[STRING]>
+ * 2: Filesystem <HASHMAP>
+ * 3: Raw path to target directory <STRING>
+ * 4: Creates a directory if it is not found <BOOL> (Optional)
```
Comment (y0014984) [inline addons/filesystem/functions/fnc_chdir.sqf:8]: In which cases, chdir is called with _create = false ?

File: addons/filesystem/functions/fnc_createDir.sqf [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/**
+ * Creates a directory.
```
Comment (y0014984) [inline addons/filesystem/functions/fnc_createDir.sqf:2]: Function isn't called anywhere.

File: addons/filesystem/functions/fnc_chdir.sqf (hunk 1 above)
Comment (GermanHydrogen) [reply addons/filesystem/functions/fnc_chdir.sqf:8]: AE3_filesystem_fnc_createFile
AE3_filesystem_fnc_createFolder

later (not implemeneted):
mkdir armaos command

**Review by y0014984 — APPROVED** [review]
Good work. I have no more annotations.

**Pull request merged**